lib/resty/etcd/v3.lua

```diff
@@ -977,23 +975,91 @@ local function watch(self, key, attr)
         filters = attr.filters and attr.filters or 0
     end
 
+    local create_request = {
+        key             = key,
+        range_end       = range_end,
+        prev_kv         = prev_kv,
+        start_revision  = start_revision,
+        watch_id        = watch_id,
+        progress_notify = progress_notify,
+        fragment        = fragment,
+        filters         = filters,
+    }
+
+    return create_request
+end
+
+
+function _grpc_M.create_grpc_watch_stream(self, key, attr, opts)
+    key = utils.get_real_key(self.key_prefix, key)
+    attr.range_end = get_range_end(key)
+
+    local req = {
+        create_request = create_watch_request(key, attr),
+    }
+
+    local conn = self.conn
+    if opts then
+        self.call_opts.timeout = opts.timeout and opts.timeout * 1000
+    end
+    if not self.call_opts.timeout then
+        self.call_opts.timeout = self.timeout * 1000
+    end
+
+    local st, err = conn:new_server_stream("etcdserverpb.Watch", "Watch", req, self.call_opts)
+    if not st then
+        return nil, err
+    end
+
+    local res, err = st:recv()
+    if not res then
+        return nil, err
+    end
+
+    return st
+end
+
+
+function _grpc_M.read_grpc_watch_stream(self, watching_stream)
```

Comment:
`read_grpc_watch_stream` is a new API, not replace the old `watch_dir`?

Reply:
Yes. We can't use watch_dir/watch_cancel directly, because the gRPC way doesn't use the HTTP long poll. It has its own model.